quantus/metrics/robustness/relative_input_stability.py

```diff
@@ -0,0 +1,309 @@
+            This is used for this __call__ only and won't be saved as attribute. If None, self.softmax is used.
+        channel_first: boolean, optional
+            Indicates of the image dimensions are channel first, or channel last.
+            Inferred from the input shape if None.
+        kwargs:
+            not used, deprecated
+        Returns
+        -------
+        relative input stability: float, np.ndarray
+            float in case `return_aggregate=True`, otherwise np.ndarray of floats
+
+
+        For each image `x`:
+         - generate `num_perturbations` perturbed `xs` in the neighborhood of `x`
+         - find `xs` which results in the same label
+         - Compute explanations `e_x` and `e_xs`
+         - Compute relative input stability objective, find max value with respect to `xs`
+         - In practise we just use `max` over a finite `xs_batch`
+        """
+        result = super().__call__(
+            model=model,
+            x_batch=x_batch,
+            y_batch=y_batch,
+            explain_func=explain_func,
+            explain_func_kwargs=explain_func_kwargs,
+            a_batch=a_batch,
+            device=device,
+            softmax=softmax,
+            channel_first=channel_first,
+            model_predict_kwargs=model_predict_kwargs,
+            s_batch=None,
+            batch_size=batch_size,
+        )
+        return list(result) if isinstance(result, Iterable) else float(result)  # noqa
+
+    def relative_input_stability_objective(
+        self, x: np.ndarray, xs: np.ndarray, e_x: np.ndarray, e_xs: np.ndarray
+    ) -> np.ndarray:
+        """
+        Computes relative input stabilities maximization objective
+        as defined here :ref:`https://arxiv.org/pdf/2203.06877.pdf` by the authors.
+
+        Parameters
+        ----------
+        x: np.ndarray
+            Batch of images.
+        xs: np.ndarray
+            Batch of perturbed images.
+        e_x: np.ndarray
+            Explanations for x.
+        e_xs: np.ndarray
+            Explanations for xs.
+
+        Returns
+        -------
+        ris_obj: np.ndarray
+            RIS maximization objective.
+        """
+        # fmt: off
+        nominator = (e_x - e_xs) / (e_x + (e_x == 0) * self._eps_min)  # prevent division by 0
+        nominator = np.linalg.norm(np.linalg.norm(nominator, axis=(-1, -2)), axis=-1)  # noqa
+        # fmt: on
+
+        denominator = x - xs
+        denominator /= x + (x == 0) * self._eps_min
+        # fmt: off
+        denominator = np.linalg.norm(np.linalg.norm(denominator, axis=(-1, -2)), axis=-1) # noqa
+        # fmt: on
+        denominator += (denominator == 0) * self._eps_min
+        return nominator / denominator
+
+    def evaluate_batch(
+        self,
+        model: ModelInterface,
+        x_batch: np.ndarray,
+        y_batch: np.ndarray,
+        a_batch: Optional[np.ndarray],
+        *args,
```

Comment:
Do we really need these? args and kwargs in signature

Reply:
Not really, but without `*args`, PyCharm complains that "Signature of method 'RelativeInputStability.evaluate_batch()' does not match signature of the base method in class 'BatchedPerturbationMetric'".

Reply:
Removed `*args`.